backend/app/modules/user/allocations/controller.py

```diff
@@ -1,13 +1,60 @@
-from typing import List, Tuple, Dict
+from typing import List, Tuple, Dict, Optional
 
 from app.context.epoch_state import EpochState
 from app.context.manager import epoch_context, state_context
-from app.exceptions import NotImplementedForGivenEpochState
-from app.modules.dto import AllocationDTO
+from app.exceptions import (
+    NotImplementedForGivenEpochState,
+    InvalidEpoch,
+    NotInDecisionWindow,
+)
+from app.modules.dto import (
+    AccountFundsDTO,
+    ProposalDonationDTO,
+    UserAllocationRequestPayload,
+    UserAllocationPayload,
+    AllocationItem,
+)
 from app.modules.registry import get_services
 from app.modules.user.allocations.service.pending import PendingUserAllocations
 
 
+def get_user_next_nonce(user_address: str) -> int:
+    service = get_services(EpochState.CURRENT).user_allocations_service
+    return service.get_user_next_nonce(user_address)
+
+
+def get_all_allocations(epoch_num: int) -> List[ProposalDonationDTO]:
+    context = epoch_context(epoch_num)
+    if context.epoch_state > EpochState.PENDING:
+        raise NotImplementedForGivenEpochState()
+    service = get_services(context.epoch_state).user_allocations_service
+    return service.get_all_allocations(context)
+
+
+def get_all_donations_by_project(
+    project_address: str, epoch_num: Optional[int] = None
+) -> List[ProposalDonationDTO]:
+    context = (
+        state_context(EpochState.PENDING)
+        if epoch_num is None
+        else epoch_context(epoch_num)
+    )
+    if context.epoch_state > EpochState.PENDING:
```

Comment:
(blocking; nitpicking / question)
not sure if we need such additional validation since we call state_context with EpochState.PENDING above?